### PR TITLE
Producer tests

### PR DESCRIPTION
**Pull request by KacperBieganek**

### CODE REVIEW

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    @Test
+    public void recordFailsToPassDataValidationWithInvalidFlightType() {
+        //given
+        String[] invalidRecord = validRecord;
+        invalidRecord[2] = "Invalid";
+        //when
+        boolean result = flightDataProducer.isDataValid(invalidRecord);
+        //then
+        assertEquals(result, false);
```
Comment (hvvka) [inline flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: Swap these arguments – they are in wrong order.
It should be: `assertEquals(expected value, real value)`.

Use `assertFalse` for this test.

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    @Test
+    public void flightObjectIsConstructedCorrectly() {
+        //given
+        Flight flight = new Flight("test", "test", FlightType.Internacional, TimeType.departureEstimate, "test",
+                FlightSituation.Realizado, "test", "test", "test", "test", "test", "test", "test", "test", "test",
+                0.0, 0.0, 0.0, 0.0);
+        //when
+        Flight result = flightDataProducer.createFlight(validRecord);
+        //then
+        assertEquals(flight, result);
```
Comment (hvvka) [inline flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: Wrong arguments' order

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    @Test
+    public void recordFailsToPassDataValidationWithInvalidFlightSituation() {
+        //given
+        String[] invalidRecord = validRecord;
+        invalidRecord[5] = "Invalid";
+        //when
+        boolean result = flightDataProducer.isDataValid(invalidRecord);
+        //then
+        assertEquals(result, false);
```
Comment (hvvka) [inline flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: Wrong arguments' order.
Use `assertFalse` for this test.

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @Test
+    public void validRecordPassesDataValidation() {
+        //given
+        //when
+        boolean result = flightDataProducer.isDataValid(validRecord);
+        //then
+        assertEquals(result, true);
```
Comment (hvvka) [inline flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: Wrong arguments' order.
Use `assertTrue` for this test.

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        //expect
+        verify(br, times(2)).readLine();
+    }
+
+    @Test
+    public void validRecordIsSentCorrectly() throws IOException {
+        //given
+        Flight flight = flightDataProducer.createFlight(validRecord);
+        Long index = 1L;
+        List<ProducerRecord<Long, Flight>> expected = Arrays.asList(
```
Comment (hvvka) [inline flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: Use `Collections.singletonList()` for a list with only one argument

File: flightdata-kafka-producer/src/main/java/com/jackdaw/Runner.java [hunk 6]
```diff
@@ -25,12 +30,17 @@ public static void main(String[] args) throws IOException {
...
         client.newCall(request).execute();
-        
-        FlightDataKafkaProducer producer = new FlightDataKafkaProducer(topicProperties.get("fileName").toString(),topicProperties.get("topicName").toString());
+        InputStream input = new FileInputStream("/volume/flightdata-kafka-producer.properties");
+        Properties props = new Properties();
+        props.load(input);
+
+        FlightDataKafkaProducer producer = new FlightDataKafkaProducer(topicProperties.get("fileName").toString(),
+                topicProperties.get("topicName").toString(),
+                new KafkaProducer<Long, Flight>(props));
```
Comment (hvvka) [inline flightdata-kafka-producer/src/main/java/com/jackdaw/Runner.java]: Can be replaced with <>

File: flightdata-kafka-producer/src/main/java/com/jackdaw/Runner.java [hunk 7]
```diff
@@ -25,12 +30,17 @@ public static void main(String[] args) throws IOException {
 
         Request request = new Request.Builder()
                 .post(RequestBody.create(SCHEMA_CONTENT, schema.toString()))
-                .url(schemaProps.getProperty("schema.registry.url").toString()+"/subjects/Flight/versions")
+                .url(schemaProps.getProperty("schema.registry.url").toString() + "/subjects/Flight/versions")
```
Comment (hvvka) [inline flightdata-kafka-producer/src/main/java/com/jackdaw/Runner.java]: `.toString()` after `.getProperty(...)` is redundant

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        invalidRecord[5] = "Invalid";
+        //when
+        boolean result = flightDataProducer.isDataValid(invalidRecord);
+        //then
+        assertEquals(result, false);
+    }
+
+    @Test
+    public void validRecordPassesDataValidation() {
+        //given
```
Comment (hvvka) [inline flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: There's nothing given, so remove this comment

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        when(br.readLine()).thenReturn(",,Internacional,departureEstimate,,Realizado,,,,,,,,,0.0,0.0,0.0,0.0")
+                .thenReturn(null);
+        //when
+        flightDataProducer.runProducer(br);
+        //expect
+        verify(br, times(2)).readLine();
+    }
+
+    @Test
+    public void validRecordIsSentCorrectly() throws IOException {
```
Comment (hvvka) [inline flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: The `IOException` is not thrown – remove it from signature

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+        flightDataProducer = new FlightDataKafkaProducer("", topic, producer);
+        validRecord = new String[]{"test", "test", "Internacional", "departureEstimate", "test", "Realizado",
+                "test", "test", "test", "test", "test", "test", "test", "test", "test",
+                "0.0", "0.0", "0.0", "0.0"};
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void ThrowsExceptionWhenStringArrayIsEmpty() {
+        //given
+        String[] emptyArray = (String[]) Array.newInstance(String.class, 0);
```
Comment (hvvka) [inline flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: This is one of the strangest empty array creation I've seen.
Simply replace this line with:
```java
String[] emptyArray = new String[0];
```
It looks confusing.

File: flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java (hunk 2 above)
Comment (KacperBieganek) [reply flightdata-kafka-producer/src/test/java/com/jackdaw/producer/FlightDataKafkaProducerTests.java]: Actually that one is in right order :smile:

**Pull request merged**